colossalai/inference/executor/rpc_worker.py

```diff
@@ -0,0 +1,294 @@
+import os
+from typing import List, Tuple, Union
+
+import rpyc
+import torch
+import torch.distributed as dist
+from torch import nn
+from transformers import AutoConfig, AutoModelForCausalLM
+from transformers.models.llama.modeling_llama import LlamaForCausalLM
+
+import colossalai
+from colossalai.accelerator import get_accelerator
+from colossalai.cluster import ProcessGroupMesh
+from colossalai.inference.config import InferenceConfig, InputMetaData
+from colossalai.inference.flash_decoding_utils import FDIntermTensors
+from colossalai.inference.modeling.policy import (
+    NoPaddingBaichuanModelInferPolicy,
+    NoPaddingLlamaModelInferPolicy,
+    model_policy_map,
+)
+from colossalai.inference.sampler import search_tokens
+from colossalai.inference.utils import get_model_size, has_index_file
+from colossalai.interface import ModelWrapper
+from colossalai.logging import get_dist_logger
+from colossalai.pipeline.stage_manager import PipelineStageManager
+from colossalai.shardformer import ShardConfig, ShardFormer
+from colossalai.shardformer.policies.base_policy import Policy
+
+PP_AXIS, TP_AXIS = 0, 1
+
+_supported_models = {
+    "LlamaForCausalLM": LlamaForCausalLM,
+    "BaichuanForCausalLM": AutoModelForCausalLM,
+}
+
+_supported_model_policies = {
+    "NoPaddingLlamaModelInferPolicy": NoPaddingLlamaModelInferPolicy,
+    "NoPaddingBaichuanModelInferPolicy": NoPaddingBaichuanModelInferPolicy,
+}
+
+logger = get_dist_logger(__name__)
+
+
+class rpcWorkerService(rpyc.Service):
+
+    """
+    Execute the computation tasks and manage its own kv cache
+
+    Func with prefix `exposed_` will be invoked by client.
+    """
+
+    def exposed_init_dist_env(self, rank, world_size, master_address, master_port):
+        logger.info(f"init process group for rank {rank}")
+        colossalai.launch(rank=rank, world_size=world_size, port=master_port, host=master_address)
+        logger.info(f"init process group done for rank {rank}")
+
+    def exposed_init_model(
+        self, inference_config_param: dict, model_or_path: Union[nn.Module, str], model_policy_param: str = None
+    ):
+        assert dist.is_initialized(), "invoke init_dist_env first please!"
+
+        self.inference_config = InferenceConfig.from_rpc_param(inference_config_param)
+        model_policy = _supported_model_policies[model_policy_param]() if model_policy_param else None
+
+        self.dtype = self.inference_config.dtype
+        self.verbose = True
+
+        self._init_model(model_or_path, model_policy)
+        self._init_fd_tensor()
+        self._init_output_tensor()
+        logger.info(f"init model done for rank {dist.get_rank()}")
+
+    def exposed_init_cache(self, alloc_shape: Tuple[Tuple[int, ...], Tuple[int, ...]]):
```

Comment:
For the latest version of dev branch, alloc shape for kcache and that for vcache could be different...

Reply:
Yeah, this is a Tuple[K_shape, V_shape], double check again :)

Reply:
Gotcha